File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/[VB]-SharePoint 2010 101 Code Samples/SharePoint 2010 Developing Solution Validators/VB/GENERAL_SolutionValidator/Features/SolutionValidatorFeature/SolutionValidatorFeature.EventReceiver.vb
Option Explicit On
Option Strict On

Imports System
Imports System.Runtime.InteropServices
Imports System.Security.Permissions
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.Administration
Imports Microsoft.SharePoint.Security
Imports Microsoft.SharePoint.UserCode

''' <summary>
''' This feature receiver installs the custom solution validator on feature activation.
''' It also cleans up on feature deactivation
''' </summary>
''' <remarks>
''' The GUID attached to this class may be used during packaging and should not be modified.
''' </remarks>
<GuidAttribute("b390b919-709f-4309-940e-4aa8ec976436")> _
Public Class SolutionValidatorFeatureEventReceiver 
    Inherits SPFeatureReceiver


    Public Overrides Sub FeatureActivated(ByVal properties As SPFeatureReceiverProperties)
        'Add the Custom Solution Validators to the Solution Validators collection
        SPUserCodeService.Local.SolutionValidators.Add(New DemoSolutionValidator(SPUserCodeService.Local))
    End Sub

    Public Overrides Sub FeatureDeactivating(ByVal properties As SPFeatureReceiverProperties)
        'Remove the Solution Validator. The Guid is the one that marks the DemoSolutionValidator class
        SPUserCodeService.Local.SolutionValidators.Remove(New Guid("93DA2F64-4438-4547-B969-226834DB9AB5"))
    End Sub

End Class

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Developing Solution Validators/C#/GENERAL_SolutionValidator/Features/SolutionValidatorFeature/SolutionValidatorFeature.EventReceiver.cs
using System;
using System.Runtime.InteropServices;
using System.Security.Permissions;
using Microsoft.SharePoint;
using Microsoft.SharePoint.Administration;
using Microsoft.SharePoint.Security;
using Microsoft.SharePoint.UserCode;

namespace GENERAL_SolutionValidator.Features.SolutionValidatorFeature
{
    /// <summary>
    /// This feature receiver installs the custom solution validator on feature activation.
    /// It also cleans up on feature deactivation
    /// </summary>
    /// <remarks>
    /// The GUID attached to this class may be used during packaging and should not be modified.
    /// </remarks>
    [Guid("308cc83e-344d-48a8-be73-6501e9de97f8")]
    public class SolutionValidatorFeatureEventReceiver : SPFeatureReceiver
    {
        public override void FeatureActivated(SPFeatureReceiverProperties properties)
        {
            //Add the Custom Solution Validators to the Solution Validators collection
            SPUserCodeService.Local.SolutionValidators.Add(new DemoSolutionValidator(SPUserCodeService.Local));
        }

        public override void FeatureDeactivating(SPFeatureReceiverProperties properties)
        {
            //Remove the Solution Validator. The Guid is the one that marks the DemoSolutionValidator class
            SPUserCodeService.Local.SolutionValidators.Remove(new Guid("CC900DAA-7D8B-4C7D-B867-5E2CF138EFB6"));
        }

    }
}
